python/ray/train/examples/torch_tensorboard_profiler_example.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+
+import torch
+from torch.nn.modules.utils import consume_prefix_in_state_dict_if_present
+
+import ray
+import ray.train as train
+from ray.train import Trainer
+from ray.train.callbacks import TBXLoggerCallback
+from ray.train.callbacks.profile import TorchTensorboardProfilerCallback, \
+    TorchWorkerProfiler
+
+
+def train_func():
+    p = TorchWorkerProfiler()
+    p.__enter__()
+
+    # Setup model.
+    model = torch.nn.Linear(1, 1)
+    model = train.torch.prepare_model(model)
+    loss_fn = torch.nn.MSELoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=1e-2)
+
+    # Setup data.
+    input = torch.randn(1000, 1)
+    labels = input * 2
+    dataset = torch.utils.data.TensorDataset(input, labels)
+    dataloader = torch.utils.data.DataLoader(dataset, batch_size=32)
+    dataloader = train.torch.prepare_data_loader(dataloader)
+
+    # Train.
+    for epoch in range(5):
+        p.record_function_enter("train_epoch")
```

Comment:
Would be great if we could make it a context manager as well, or just use PyTorch's `record_function` directly (if possible)